nilearn/datasets/func.py

```diff
@@ -1640,3 +1640,176 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
+                                      'Dominant Hand', 'Sex'],
+                               delimiter=',', dtype=['U9', '<f8',
+                                                     'U1', 'U1'])
+
+    # Keep phenotypic information for selected subjects
+    int_ids = np.asarray(ids)
+    phenotypic = phenotypic[[np.where(phenotypic['Subject'] == i)[0][0]
+                             for i in int_ids]]
+
+    # Download fsaverage surfaces and sulcal information
+    surf_file = os.path.join('fsaverage5', '%s.%s.gii')
+    surf_url = url + '%i/%s.%s.gii'
+    surf_nids = {'lh pial': 8465, 'rh pial': 8468,
+                 'lh infl': 8464, 'rh infl': 8467,
+                 'lh sulc': 8466, 'rh sulc': 8469}
+
+    pials = []
+    infls = []
+    sulcs = []
+    for hemi in [('lh', 'left'), ('rh', 'right')]:
+
+        pial = _fetch_files(data_dir,
+                            [(surf_file % ('pial', hemi[1]),
+                             surf_url % (surf_nids['%s pial' % hemi[0]],
+                                         hemi[0], 'pial'),
+                             {'move': surf_file % ('pial', hemi[1])})],
+                            resume=resume, verbose=verbose)
+        pials.append(pial)
+
+        infl = _fetch_files(data_dir,
+                            [(surf_file % ('pial_inflated', hemi[1]),
+                             surf_url % (surf_nids['%s infl' % hemi[0]],
+                                         hemi[0], 'inflated'),
+                             {'move': surf_file % ('pial_inflated', hemi[1])})],
+                            resume=resume, verbose=verbose)
+        infls.append(infl)
+
+        sulc = _fetch_files(data_dir,
+                            [(surf_file % ('sulc', hemi[1]),
+                             surf_url % (surf_nids['%s sulc' % hemi[0]],
+                                         hemi[0], 'sulc'),
+                             {'move': surf_file % ('sulc', hemi[1])})],
+                            resume=resume, verbose=verbose)
+        sulcs.append(sulc)
+
+    # Download subjects' datasets
+    func_right = []
+    func_left = []
+    for i in range(len(ids)):
+
+        archive = url + '%i/%s_%s_preprocessed_fsaverage5_fwhm6.gii'
+        func = os.path.join('%s', '%s_%s_preprocessed_fwhm6.gii')
+        rh = _fetch_files(data_dir,
+                          [(func % (ids[i], ids[i], 'right'),
+                           archive % (nitrc_ids[i], ids[i], 'rh'),
+                           {'move': func % (ids[i], ids[i], 'right')}
+                            )],
+                          resume=resume, verbose=verbose)
+        lh = _fetch_files(data_dir,
+                          [(func % (ids[i], ids[i], 'left'),
+                           archive % (nitrc_ids[i], ids[i], 'lh'),
+                           {'move': func % (ids[i], ids[i], 'left')}
+                            )],
+                          resume=resume, verbose=verbose)
```

Comment:
This usage of the low_level "_fetch_files" function is telling me that we probably need an additional data fetcher, to answer this specific usecase.